binaries/upload.py

```diff
@@ -4,71 +4,58 @@
 import os
 import sys
 
+REPO_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
+sys.path.append(REPO_ROOT)
+
+from ts_scripts.utils import try_and_handle
+
 # The following environment variables are expected to be populated in the shell environment
+# TODO: Where are these 4 parametrs actually used?
 PYPI_USERNAME_ENV_VARIABLE = "TWINE_USERNAME"
 PYPI_PASSWORD_ENV_VARIABLE = "TWINE_PASSWORD"
 
 CONDA_TOKEN_ENV_VARIABLE = "CONDA_TOKEN"
 CONDA_USER = "torchserve-staging"
 
-REPO_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
-
-CONDA_PACKAGES_PATH = os.path.join(REPO_ROOT, "binaries", "conda", "output")
-
-TS_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "dist"))[0]
-MA_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "model-archiver", "dist"))[0]
-WA_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "workflow-archiver", "dist"))[0]
-
 
-PACKAGES = ["torchserve", "model-archiver", "workflow-archiver"]
-
-
-def upload_pypi_packages(args):
+def upload_pypi_packages(args, WHL_PATHS):
     """
     Takes a list of path values and uploads them to pypi using twine, using token stored in environment variable
     """
+    dry_run = args.dry_run
 
     # Note: TWINE_USERNAME and TWINE_PASSWORD are expected to be set in the environment
-    for dist_path in [TS_WHEEL_PATH, MA_WHEEL_PATH, WA_WHEEL_PATH]:
+    for dist_path in WHL_PATHS:
         if args.test_pypi:
-            exit_code = os.system(
-                f"set -ex ; twine upload {dist_path}/* --username __token__ --repository-url https://test.pypi.org/legacy/"
+            try_and_handle(
+                f"twine upload {dist_path}/* --username __token__ --repository-url https://test.pypi.org/legacy/",
+                dry_run,
             )
         else:
-            exit_code = os.system(
-                f"set -ex ; twine upload --username __token__ {dist_path}/*"
-            )
-        if exit_code != 0:
-            sys.exit(f"twine upload for path {dist_path} failed")
+            try_and_handle(f"twine upload --username __token__ {dist_path}/*", dry_run)
 
-    if args.test_pypi:
-        print(
-            f"All packages uploaded to test.pypi.org successfully. Please install package as 'pip install -i https://test.pypi.org/simple/ <package-name>'"
-        )
 
-
-def upload_conda_packages():
+# TODO: to make dry_run work need to mock a fake os path for CONDA_PACKAGES_PATH
+def upload_conda_packages(args, PACKAGES, CONDA_PACKAGES_PATH):
     """
     Takes a list of path values and uploads them to anaconda.org using conda upload, using token stored in environment variable
+    If you'd like to upload to a staging environment make sure to pass in your personal credentials when you anaconda login instead
+    of the pytorch credentials
```

Comment:
One simple trick I am using is that in my fork, I use the same token name, but its linked to my repo

Reply:
Yeah reminds me that I haven't tested the action end to end yet - just tested the individual components